backpack/custom_module/graph_utils.py

```diff
@@ -0,0 +1,203 @@
+    graph: Graph = BackpackTracer().trace(module)
+    for node in graph.nodes:
+        if node.op == "call_function":
+            if str(node.target) == "<built-in function mul>":
+                _change_node_to_module(
+                    node,
+                    "scale_module",
+                    module,
+                    ScaleModule(node.args[1]),
+                    (node.args[0],),
+                )
+            counter += 1
+
+    graph.lint()
+    print(f"\tMultiplications transformed: {counter}")
+    return GraphModule(module, graph)
+
+
+def _transform_add_to_sum_module(module: Module) -> GraphModule:
+    print("\tBegin transformation: <built-in function add> -> SumModule")
+    counter: int = 0
+    graph: Graph = BackpackTracer().trace(module)
+    for node in graph.nodes:
+        if node.op == "call_function":
+            if str(node.target) == "<built-in function add>":
+                _change_node_to_module(
+                    node, "sum_module", module, SumModule(), node.args
+                )
+                counter += 1
+
+    graph.lint()
+    print(f"\tSummations transformed: {counter}")
+    return GraphModule(module, graph)
+
+
+def _transform_flatten_to_module(module: Module) -> GraphModule:
+    print("\tBegin transformation: <built-in method flatten> -> Flatten")
+    counter: int = 0
+    graph: Graph = BackpackTracer().trace(module)
+    for node in graph.nodes:
+        if node.op == "call_function":
+            if "<built-in method flatten" in str(node.target):
+                start_dim = node.args[1] if len(node.args) > 1 else 1
+                end_dim = node.args[2] if len(node.args) > 2 else -1
+                _change_node_to_module(
+                    node,
+                    "flatten",
+                    module,
+                    Flatten(start_dim, end_dim),
+                    (node.args[0],),
+                )
+                counter += 1
+
+    graph.lint()
+    print(f"\tFlatten transformed: {counter}")
+    return GraphModule(module, graph)
+
+
+def _transform_inplace_to_normal(
+    module: Module, initialize_recursion: bool = True
+) -> Module:
+    if initialize_recursion:
+        print("\tBegin transformation: in-place -> standard")
+        _transform_inplace_to_normal.counter = 0
+    if hasattr(module, "inplace") and module.inplace:
+        module.inplace = False
+        _transform_inplace_to_normal.counter += 1
+    for child_module in module.children():
+        _transform_inplace_to_normal(child_module, initialize_recursion=False)
+
+    if initialize_recursion:
+        print(f"\tIn-place changed: {_transform_inplace_to_normal.counter}")
+        delattr(_transform_inplace_to_normal, "counter")
+    return module
+
+
+def _transform_remove_duplicates(module: Module, max_depth: int = 100) -> GraphModule:
+    print("\tBegin transformation: remove duplicates")
+    counter = 0
+    graph: Graph = BackpackTracer().trace(module)
+    targets: Set[str] = set()
```

Comment:
`targets = set(node.target for node in graph.nodes)`

Reply:
The purpose is different. I renamed the variable to `targets_visited`.
If clear resolve. If not, maybe `targets_already_visited` is better?